helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -1261,31 +1258,62 @@ private void processDataOrChildChange(WatchedEvent event, long notificationTime)
 
     if (event.getType() == EventType.NodeDataChanged || event.getType() == EventType.NodeDeleted
         || event.getType() == EventType.NodeCreated) {
+      boolean isPathExist = event.getType() != EventType.NodeDeleted;
       Set<IZkDataListenerEntry> listeners = _dataListener.get(path);
       if (listeners != null && !listeners.isEmpty()) {
-        fireDataChangedEvents(event.getPath(), listeners, OptionalLong.of(notificationTime));
+        fireDataChangedEvents(event.getPath(), listeners, notificationTime, isPathExist);
       }
     }
   }
 
+  private void fireDataChangedEvents(final String path, Set<IZkDataListenerEntry> listeners) {
```

Comment:
Could you add some javadoc on how this is different from the other method?

Reply:
done